### PR TITLE
Support using private S3 buckets as remote mvn deployment targets

### PR DESCRIPTION
**Pull request by RickMoynihan**
Works via S3 wagon.

This PR builds on and includes commits from #18 which also adds exec-fn support.

### CODE REVIEW

File: src/deps_deploy/deps_deploy.clj [hunk 1]
```diff
@@ -6,6 +6,11 @@
             [clojure.java.io :as io]
             [clojure.data.xml :as xml]))
 
+
+(aether/register-wagon-factory! "s3p" #(org.springframework.build.aws.maven.PrivateS3Wagon.))
```
Comment (slipset) [inline src/deps_deploy/deps_deploy.clj:10]: I think I'd like for this to be `:imported` so we can say 
`((aether/register-wagon-factory! "s3p" #(PrivateS3Wagon.))`
Ditto for the following line.

**Review by slipset — CHANGES_REQUESTED** [review]
Unless there are reasons for not using `:import` I'd rather you imported the Wagon stuff

**Comment by RickMoynihan** [comment]
Agreed.  Will try and update that later.

**Comment by RickMoynihan** [comment]
I've pushed some changes to this.  Also I should probably add some docs to describe how to set this up.  I myself had made some mistakes in my local configuration that led me to assume it was doing one thing when instead it was doing another (picking up credentials via the AWS credential provider chain).

**Comment by slipset** [comment]
Sorry for dropping the ball on this, but yes, please add some docs on this in the readme

**Comment by nnichols** [comment]
Is there any way I could help w/ this? I'd love to be able to use this functionality

**Comment by slipset** [comment]
I'll ping @RickMoynihan to see if he has some docs to add to the readme.

**Comment by RickMoynihan** [comment]
Hi,

Sorry I've been off on paternity leave followed by xmas and new year, and now trying to catch up on everything that's happened at work etc.  I'll try and get round to this as soon as I can but no promises as to when.

**Comment by RickMoynihan** [comment]
To briefly summarise what you need to do to use this, you just need to provide the credentials via the AWS provider chain.  Which more or less means either:

1. setting the env vars: `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`
2. providing them via java system properties `aws.accessKeyId` and `aws.secretKey`

or 

3. via an AWS credential profile, in the file `~/.aws/credentials` with the `AWS_PROIFLE` env var used to specify which profile to use (or the 'default' profile).

```
[default]
aws_access_key_id = AKIAXXXXX
aws_secret_access_key = SECRET_KEY
```

For more details see [s3-wagon-provider](https://github.com/s3-wagon-private/s3-wagon-private#aws-credential-providers) and if you need to know how to [configure an S3 bucket see here](https://github.com/s3-wagon-private/s3-wagon-private#aws-policy).

**Pull request merged**